packages/components/src/popover/stories/index.story.tsx

```diff
@@ -255,3 +259,90 @@ export const WithSlotOutsideIframe: StoryObj< typeof Popover > = {
 		...Default.args,
 	},
 };
+
+export const WithCloseHandlers: StoryObj< typeof Popover > = {
+	render: function WithCloseHandlersStory( args ) {
+		const [ isVisible, setIsVisible ] = useState( false );
+		const buttonRef = useRef< HTMLButtonElement >( null );
+
+		const toggleVisible = ( event: React.MouseEvent ) => {
+			if ( buttonRef.current && event.target !== buttonRef.current ) {
+				return;
+			}
+			setIsVisible( ( prev ) => ! prev );
+		};
+
+		const handleClose = () => {
+			args.onClose?.();
+			setIsVisible( false );
+		};
+
+		const handleFocusOutside = ( e: SyntheticEvent ) => {
+			args.onFocusOutside?.( e );
+			setIsVisible( false );
+		};
+
+		useEffect( () => {
+			buttonRef.current?.scrollIntoView( {
+				block: 'center',
+				inline: 'center',
+			} );
+		}, [] );
+
+		return (
+			<div
+				style={ {
+					width: '300vw',
+					height: '300vh',
+					display: 'flex',
+					alignItems: 'center',
+					justifyContent: 'center',
+				} }
+			>
+				<Button
+					variant="secondary"
+					onClick={ toggleVisible }
+					ref={ buttonRef }
+				>
+					Toggle Popover
+					{ isVisible && (
+						<Popover
+							{ ...args }
+							onClose={ handleClose }
+							onFocusOutside={ handleFocusOutside }
+						>
+							{ args.children }
+						</Popover>
+					) }
+				</Button>
+			</div>
+		);
+	},
+	args: {
+		...Default.args,
+		children: (
+			<div style={ { width: '280px', whiteSpace: 'normal' } }>
+				<div style={ { padding: '16px' } }>
+					<p>
+						This popover demonstrates proper use of event handlers:
+					</p>
+					<ul className="mb-2 list-disc pl-4">
+						<li>
+							<strong>onClose</strong>: Called when escape key is
+							pressed
+						</li>
+						<li>
+							<strong>onFocusOutside</strong>: Called when focus
+							moves outside the popover
+						</li>
+					</ul>
+					<p>
+						Try clicking outside or pressing ESC to close this
+						popover.
+					</p>
+					<Button className="mt-2">Focus me then tab away</Button>
+				</div>
+			</div>
```

Comment:
I feel like we can simplify the contents to something like this:

```suggestion
			<div style={ { width: '280px', whiteSpace: 'normal' } }>
				<p>
					Clicking outside triggers the onFocusOutside callback prop.
					Pressing the Escape key triggers the onClose callback prop.
				</p>
			</div>
```

Let's also pass `focusOnMount: true` to the `args` to make sure the `Popover` receives focus when opened

Reply:
Yes, this looks much cleaner! 
I have split it into two paragraphs for better view 👍🏻